### PR TITLE
Refactored hook handlers in mod_domain_isolation

### PR DESCRIPTION
**Pull request by pawlooss1**
This PR changes all hook handlers in `mod_domain_isolation` module to `gen_hook` format. Also, I introduced a minor refactor and added missing `spec` for private functions.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/150372) / small_tests / 22d7b49e9070f154fee714de1018f546dde93163
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3848/150372/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3848/150372/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-09_10.00.49/lib.mongooseim.logs/run.2022-11-09_10.00.49/suite.log.html)

File: src/mod_domain_isolation.erl [hunk 1]
```diff
@@ -11,64 +11,73 @@
...
-filter_local_packet({#jid{lserver = Server},
-                     #jid{lserver = Server}, _Acc, _Packet} = Arg) ->
-    Arg;
-filter_local_packet({#jid{lserver = FromServer} = From,
-                     #jid{lserver = ToServer} = To, Acc, _Packet} = Arg) ->
+-spec filter_local_packet(drop, _, _) -> {ok, drop};
+                         (FPacketAcc, Params, Extra) -> {ok, FPacketAcc} when
+      FPacketAcc :: mongoose_hooks:filter_packet_acc(),
+      Params :: map(),
+      Extra :: map().
```
Comment (NelsonVides) [inline src/mod_domain_isolation.erl:37]: Likewise, `gen_hook:extra()`

**Review by NelsonVides — COMMENTED** [review]
Same small comments as for many hooks.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/150958) / small_tests / 9370c3d7af70803c13037679538ce65e713eba21
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3848/150958/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3848/150958/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-10_09.19.05/lib.mongooseim.logs/run.2022-11-10_09.19.05/suite.log.html)

**Review by NelsonVides — APPROVED** [review]
LGTM 👌🏽

**Pull request merged**